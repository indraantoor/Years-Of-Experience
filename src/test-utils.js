import { Provider } from "react-redux";
import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { userDetailsSlice } from "./store/userDetailsSlice";
import { workExperiencesSlice } from "./store/workExperiencesSlice";
import { StaticRouter } from "react-router-dom/server";

function getStoreWithState(state) {
  const store = configureStore({
    reducer: {
      userDetails: userDetailsSlice.reducer,
      workExperiences: workExperiencesSlice.reducer,
    },
    state,
  });
  return store;
}

export function renderWithContext(children, state) {
  const store = getStoreWithState(state);
  render(
    <StaticRouter>
      <Provider store={store}>{children}</Provider>
    </StaticRouter>
  );

  return { store };
}

export function getStateWithItems(userDetails, workExperiences) {
  const state = {
    userDetails: {
      loading: true,
      error: false,
      redirect: false,
    },
    workExperiences: {
      data: [],
      loading: true,
      error: false,
      redirect: false,
    },
  };
  return state;
}
