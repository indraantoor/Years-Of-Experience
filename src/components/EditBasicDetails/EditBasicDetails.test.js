/**
 *  @jest-environment jsdom
 */
import React from "react";
import { fireEvent, screen } from "@testing-library/react";
import { EditBasicDetails } from ".";
import "@testing-library/jest-dom";
import { StaticRouter } from "react-router-dom/server";
import { create } from "react-test-renderer";
import { renderWithContext } from "../../test-utils";

import { Provider } from "react-redux";

jest.mock("@firebase/firestore", () => ({
  doc: jest.fn(),
  getDoc: jest.fn(),
  updatedoc: jest.fn(),
  initializeApp: jest.fn(),
  getFirestore: jest.fn(),
  enableIndexedDbPersistence: jest.fn(),
}));

jest.mock("@firebase/storage", () => ({
  ref: jest.fn(),
  uploadBytesResumable: jest.fn(),
  getDownloadURL: jest.fn(),
}));

jest.mock("../../firebase-config.js", () => ({
  getStorage: jest.fn(),
}));

jest.mock("../../store/requests.js", () => ({
  updateUserDetailsRequest: jest.fn(),
}));

test("renders correctly", () => {
  const { store } = renderWithContext(<EditBasicDetails />);

  const tree = create(
    <StaticRouter>
      <Provider store={store}>
        <EditBasicDetails />
      </Provider>
    </StaticRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test("username has no spaces", () => {
  renderWithContext(<EditBasicDetails />);

  function hasInputValue(e, inputValue) {
    return screen.getByDisplayValue(inputValue) === e;
  }

  const inputValue = "user name";
  const usernameInput = screen.getByLabelText("Username");
  fireEvent.change(usernameInput, { target: { value: inputValue } });

  expect(hasInputValue(usernameInput, "username")).toBe(true);
});
