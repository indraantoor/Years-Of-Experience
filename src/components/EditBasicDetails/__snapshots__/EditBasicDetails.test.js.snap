// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ gpsELF"
>
  <form
    className="sc-gsnTZi iCcdGT"
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name
      <input
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
      />
    </label>
    <label
      htmlFor="username"
    >
      Username
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
      />
    </label>
    <label
      htmlFor="age"
    >
      Age
      <input
        id="age"
        max="100"
        min="1"
        name="age"
        onChange={[Function]}
        placeholder="Age"
        type="number"
      />
    </label>
    <div
      className="submitBtnContainer"
    >
      <button
        data-testid="submitBtn"
        type="submit"
      >
        Update
      </button>
    </div>
  </form>
</div>
`;
